shell/platform/darwin/ios/framework/Source/FlutterPlugins.mm

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/framework/Headers/FlutterPlugins.h"
+
+@implementation FlutterMethodPlugin
+- (instancetype)initWithController:(FlutterViewController *)controller
+                       channelName:(NSString*)channelName {
+  NSAssert(channelName, @"Channel name cannot be nil");
+  return [self initWithController:controller
+                          channel:[FlutterMethodChannel methodChannelWithName:channelName
+                                                              binaryMessenger:controller]];
+}
+
+- (instancetype)initWithController:(FlutterViewController *)controller
+                           channel:(FlutterMethodChannel*)channel {
+  NSAssert(controller, @"Controller cannot be nil");
+  NSAssert(channel, @"Channel cannot be nil");
+  self = [super init];
+  NSAssert(self, @"Super init cannot be nil");
+  _controller = controller;
+  _channel = channel;
+  __unsafe_unretained id weakSelf = self;
```

Comment:
In case of ARC, you would be able to use `__weak`. I think `__unsafe_unretained` would only be useful if you had to target old iOS versions (~ iOS 5 if memory serves). `__weak` wasn't available then.